design/pipeline-artifacts.md

```diff
@@ -1,30 +1,30 @@
+Pipeline Artifacts are the new way to move files between jobs and stages in your pipeline. They are hosted in Azure Artifacts and will eventually entirely replace FCS "Build Artifacts". Because moving files between jobs and stages is a crucial part of most CI/CD workflows, and because Pipeline Artifacts are expected to be the default way to do so, this spec proposes a YAML shortcut syntax for uploading and downloading artifacts.
 
-In this document, `artifact` refers specifically to publishing Pipeline Artifacts from the current pipeline. `downloadArtifact` refers to downloading Pipeline Artifact artifacts from the current pipeline, from other Azure Pipelines, and from other pipeline resources (e.g. Jenkins builds). 
+In this document, `artifact` refers specifically to uploading Pipeline Artifacts from the current pipeline. `download` refers to downloading Pipeline Artifact artifacts from the current pipeline and from other Azure Pipelines. Other pipeline systems (e.g. Jenkins) will be handled as [pipeline resources](pipeline-resources.md).
 
-Artifacts are distinct from other `resources` types, including `containers`, `repositories`, and `feeds`.
+Artifacts are distinct from other `resources` types, including `containers`, `repositories`, `packages`, and `feeds`.
 
-## Publishing artifacts: `artifact`
+## Uploading artifacts: `upload`
 
-`artifact` is a shortcut for the [Publish Pipeline Artifacts](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/utility/publish-pipeline-artifact.md) task. It will publish files from the current job to be used in subsequent jobs or in other stages or pipelines.
+`upload` is a shortcut for the [Upload Pipeline Artifact](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/utility/publish-pipeline-artifact.md) task (formerly called "Publish Pipeline Artifacts"). It will upload files from the current job to be used in subsequent jobs or in other stages or pipelines.
 
 ### Schema
 
 ```yaml
-- artifact: string | [ string ]  # a minimatch path or list of [minimatch paths](tasks/file-matching-patterns.md) to publish
+- upload: string | [ string ]  # a minimatch path or list of [minimatch paths](tasks/file-matching-patterns.md) to upload
   name: string # identifier for this artifact (no spaces allowed), defaults to 'default'
-  prependPath: string # a directory path that will be prepended to all published files
+  prependPath: string # a directory path that will be prepended to all uploaded files
   seal: boolean # if true, finalizes the artifact so no more files can be added after this step
 ```
 
 ### Example
 
 ```yaml
-- artifact:
+- upload:
```

Comment:
We considered that option but went with what's currently in the spec as it allows for a one-liner in the simple case: `- upload: <my file selector>`. (This would upload files to the `default` artifact.) I'm open to discussions on consistency though. @vtbassmatt your thoughts?

Reply:
That's fine if you want to optimize for the selector. But I do think the property that denotes the artifact name should be `artifact` or `artifactName`. The problem is `name` is a core property on almost all of the steps. It means the "task reference name" and is used when programmatically accessing the output variables set by the task.

Reply:
Ah, makes total sense. Changed to `artifact`.